plugins/org.locationtech.udig.project.edit/src/org/locationtech/udig/project/internal/provider/ProjectItemProviderAdapterFactory.java

```diff
@@ -105,21 +105,10 @@ public Adapter createContextModelAdapter() {
      */
     @Override
     public Adapter createLayerAdapter() {
-        if (layerItemProvider == null) {
-            layerItemProvider = new LayerItemProvider(this);
-        }
-
-        return layerItemProvider;
+        return new LayerItemProvider(this);
```

Comment:
This is an interesting change - I guess it makes a unique provider each time (rather than a single one). This is only necessary if the provider is now statefull - is that now the case?